### PR TITLE
refactor: migrate config parsing from Babel AST to jiti

### PR DESCRIPTION
**Pull request by mantrakp04**
## Summary

Replace `parseHexclaveConfigFileContent` / `evaluateStaticConfigExpression` (Babel AST walker) with `evalConfigFileContent` using `jiti.evalModule()`. Move `renderConfigFileContent` from `hexclave-config-file.ts` → `config-rendering.ts`.

Added `jiti` dep to `@hexclave/shared` (already used in shared-backend, dashboard, backend, cli).

Link to Devin session: https://app.devin.ai/sessions/cb098b1fb62b4dfeaf3324bc2e1377f1
Requested by: @mantrakp04

---
## Summary by cubic
Migrates trusted config evaluation to `jiti` and moves GitHub config edits to a server‑side repo agent running in a Vercel Sandbox with an apply → review → commit flow. Adds run tracking, safer defaults, and a dashboard diff review with clear, user‑facing errors.

- **New Features**
  - Two‑phase flow and endpoints: POST `/internal/config/github/apply`, `.../commit`, `.../cancel`, plus GET `.../run`; each run tracked by `run_id` in `ConfigAgentRun` (status, stage, progress, diff, base commit, sandbox id). Run ids validated as UUIDs.
  - Repo agent runs in a fresh sandboxed clone; warm‑boot via base snapshot (`apps/backend/scripts/config-agent/build-image.ts`, `HEXCLAVE_CONFIG_AGENT_BASE_SNAPSHOT_ID`). Captures a unified diff and base commit, stops the sandbox at review, then rebuilds files from the stored diff on commit. Returns `commitSha`, uses a safe conflict error, and strips OAuth tokens from git remotes.
  - Dashboard: non‑dismissible progress and diff preview using `@pierre/diffs` with a cross‑tab run watcher; blocks conflicting edits and supports cancel/commit review flow. Adds an RDE “apply” path with progress UI.
  - AI proxy defaults to `/api/latest/integrations/ai-proxy` (production passthrough via `PRODUCTION_AI_PROXY_BASE_URL`); adds `anthropic/claude-haiku-4.5`.

- **Refactors and Fixes**
  - Trusted eval via `@hexclave/shared` `config-eval` using `jiti`; browser‑safe parsing for untrusted GitHub content; rendering remains in `config-rendering`. Clear separation of Node‑only code into `config-eval`.
  - Shared agent/updater logic moved to `@hexclave/shared-backend`; removed deterministic fast path so all writes go through the agent to preserve authoring. CLI and emulator updated to use `config-eval`.
  - Defaults/renames: config file `hexclave.config.ts` (CLI `config pull` defaults to this path), workflow `hexclave-config-sync.yml`; env prefixes standardized to `HEXCLAVE_*`.
  - Integrity and UX: commit advancement gated to the current linked repo/branch; cancel clears any captured diff; elapsed timer handles late starts and the not‑started sentinel; loader vs invalid config export errors separated for accurate messaging.
  - Onboarding and seeds: wizard now uses environment‑based OAuth provider setup with updated tests; corrected GitHub owner in dummy project seeding.

<sup>Written for commit 6cf0e899a06be149da4d6a5d125b256e5e4f619a. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/hexclave/hexclave/pull/1661?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Refactor**
  * Improved configuration file parsing/validation by evaluating config modules, supporting both string and object-based `config` exports and ensuring the expected `config` export is present.
  * Updated config rendering and import-package detection to consistently generate the `config` export and handle legacy package entrypoints.
  * Tightened handling of non-statically-resolvable forms during update flows.
* **Tests**
  * Updated and extended config parsing/validation tests to reflect the new evaluation behavior and edge cases.
* **Chores**
  * Added a Jiti-based dependency to support runtime evaluation.

### CODE REVIEW

**Comment by mantrakp04** [comment]
@greptile-ai review

File: apps/backend/src/app/api/latest/internal/config/github/apply/route.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+          });
+        } else {
+          // Transition to awaiting_review — sandbox stays alive.
+          await setConfigAgentRunAwaitingReview({
+            projectId,
+            branchId,
+            diff: result.diff,
+          });
+        }
+      } catch (error) {
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/app/api/latest/internal/config/github/apply/route.tsx:130]: Fixed in 212502a0: the apply route now stores the booted sandbox id from onSandboxId and calls stopConfigAgentSandbox in the background catch path before recording the terminal error, so a post-boot apply failure does not leave the sandbox running.

**Comment by mantrakp04** [comment]
@greptile-ai review

**Comment by mantrakp04** [comment]
@greptile-ai review

File: packages/shared/src/config-rendering.ts [hunk 2]
```diff
@@ -98,27 +160,22 @@ import.meta.vitest?.test("parseHexclaveConfigFileContent parses static config ex
...
 
-import.meta.vitest?.test("tryParseHexclaveConfigFileContent returns null for non-static exports", ({ expect }) => {
-  // Wrapped in a helper call (e.g. defineStackConfig) -> not a plain literal.
-  expect(tryParseHexclaveConfigFileContent("export const config = makeConfig();", "stack.config.ts")).toBeNull();
-  // References an imported value -> has structure to preserve.
-  expect(tryParseHexclaveConfigFileContent('import x from "./x.txt" with { type: "text" };\nexport const config = { a: x };', "stack.config.ts")).toBeNull();
-  // Syntax error.
-  expect(tryParseHexclaveConfigFileContent("export const config = {", "stack.config.ts")).toBeNull();
+import.meta.vitest?.test("tryEvalConfigFileContent returns null on failure", ({ expect }) => {
+  expect(tryEvalConfigFileContent("export const config = {", "stack.config.ts")).toBeNull();
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply packages/shared/src/config-rendering.ts]: Good catch on the spirit of this. After the jiti migration the old "non-static exports" test got split up: the factory-call and missing-import cases are already covered, but the malformed/syntax-error case had become untested at every level. Added two tests for it — a unit test in `config-eval.ts` (eval throws on invalid syntax) and a caller-level test in `config-file.test.ts` (an unparsable config rejects with "Failed to load config file"). Both pass. ✅ Fixed.

File: apps/backend/src/lib/config/repo-agent.tsx [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    finished.stdout().catch(() => ""),
+    finished.stderr().catch(() => ""),
+  ]);
+  return { exitCode: finished.exitCode, stdout, stderr };
+}
+
+async function run(sandbox: Sandbox, cmd: string, args: string[], opts?: { cwd?: string, env?: Record<string, string>, sudo?: boolean }): Promise<RunResult> {
+  const r = await runRaw(sandbox, cmd, args, opts);
+  if (r.exitCode !== 0) {
+    throw new ConfigRepoAgentError(`Command failed (exit ${r.exitCode}): ${cmd} ${args.join(" ")}\n${(r.stderr || r.stdout).slice(-1500)}`);
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/lib/config/repo-agent.tsx]: Not an issue on the current code (the snippet here is from an earlier commit). The `run()` helper now wraps its **entire** error message in `redactTokens(...)` before throwing, which scrubs the `x-access-token:<token>@` segment — and that's the only place a token is ever embedded (the clone URL). Reproduced a transient clone failure and confirmed the thrown message contains no token across all GitHub token formats. Resolving as outdated.

File: apps/backend/src/lib/config/repo-agent.tsx [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+async function installDeps(sandbox: Sandbox, pm: string): Promise<void> {
+  if (pm !== "npm") await run(sandbox, "corepack", ["enable"], { sudo: true });
+  await run(sandbox, pm, ["install"], { cwd: REPO_DIR });
+}
+
+// ---------------------------------------------------------------------------
+// Agent
+// ---------------------------------------------------------------------------
+
+function flattenOverride(update: EnvironmentConfigOverrideOverride): Array<{ path: string, value: unknown }> {
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/lib/config/repo-agent.tsx]: Outdated — `flattenOverride` no longer exists on this branch. The flow was rewritten to send the **full** config object to the agent as JSON ("must end up deep-equal to this value"), and any dot-notation deltas are expanded into the nested object upstream before that. So nested objects are preserved exactly; there's no shallow flatten to lose data. Resolving.

File: apps/backend/src/lib/config/repo-agent.tsx [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+    token: creds.token,
+  });
+
+  let result: ConfigUpdatePushResult;
+  let newSnapshot: ConfigRepoSnapshot;
+  try {
+    await ensureTls(sandbox);
+    // Pull the latest base branch (token injected, not persisted).
+    await run(sandbox, "git", ["-C", REPO_DIR, "checkout", ref.branch]);
+    await run(sandbox, "git", ["-C", REPO_DIR, "pull", tokenUrl(githubToken, ref), ref.branch]);
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/lib/config/repo-agent.tsx]: Doesn't apply to this design. There's no "snapshot update mode" that pulls into an existing clone — every run does a fresh shallow clone, the base snapshot contains no repo and no `node_modules`, and there is no in-sandbox typecheck/build gate at all (validation runs via the linked repo's own GitHub Action after the commit is pushed). With nothing installed and no gate, the stale-`node_modules` failure can't occur. Resolving as outdated.

File: apps/backend/src/lib/config/index.tsx [hunk 6]
```diff
@@ -475,6 +475,282 @@ export async function unlinkBranchConfigOverrideSource(options: {
...
+  configUpdate: Record<string, unknown>,
+}): Promise<Record<string, unknown>> {
+  const current = await rawQuery(globalPrismaClient, getBranchConfigOverrideQuery({ projectId: options.projectId, branchId: options.branchId }));
+  const merged = override(current as Config, options.configUpdate as Config);
+  // Dashboard saves usually arrive as dot-notation deltas (for example
+  // `auth.allowSignUp: false`). The branch override can be empty, so missing
+  // parents must be materialized instead of silently dropping the pending edit.
+  return normalize(merged, { onDotIntoNonObject: "ignore", onDotIntoNull: "empty-object" }) as Record<string, unknown>;
+}
+
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/lib/config/index.tsx:548]: Outdated — `tryStartConfigAgentRun` and the single mutable per-branch record it describes don't exist here. `ConfigAgentRun` is an append-only table (one UUID row per run); `startConfigAgentRun` always inserts a new row and never overwrites a prior one, so a concurrent `/apply` can't drop the earlier run's `sandbox_id` or orphan its sandbox — the old run stays a distinct, independently cancellable row (cancel handles both `running` and `awaiting_review`). Non-serialization is intentional, and there's a test asserting concurrent runs coexist. Resolving.

File: apps/backend/src/lib/config/index.tsx [hunk 7]
```diff
@@ -475,6 +475,282 @@ export async function unlinkBranchConfigOverrideSource(options: {
...
+      FOR UPDATE
+    `;
+    const source = rows[0]?.source;
+    if (!source || source.type !== "pushed-from-github") {
+      throw new HexclaveAssertionError("Config source is not linked to GitHub; cannot run the config agent.");
+    }
+    const run = source.agent_run;
+    if (run && run.status === "running" && options.nowMs - run.started_at < options.staleMs) {
+      return { started: false, source: source as GithubConfigSource };
+    }
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/lib/config/index.tsx]: Same root cause as the other Greptile thread on this file — outdated. `tryStartConfigAgentRun` / the single overwritable per-branch `agent_run` record no longer exist. `ConfigAgentRun` is append-only (one UUID row per run); a concurrent start inserts a new row and can't overwrite or orphan the prior run's sandbox. The suggested `awaiting_review` guard targets code that isn't here. Resolving.

File: apps/backend/src/app/api/latest/internal/config/github/commit/route.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    }
+    const runStartedAt = run.started_at;
+
+    const sandboxId = run.sandbox_id;
+    if (!sandboxId) {
+      // Sandbox id was never recorded — can't commit. Treat as an error and clean up.
+      await recordConfigAgentRunResult({
+        projectId,
+        branchId,
+        runStartedAt,
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/app/api/latest/internal/config/github/commit/route.tsx]: Outdated — the idempotent "sandbox-expired" response this refers to was removed in the latest commit. The commit flow no longer replays from a live sandbox or checks `sandbox_id`; it replays a persisted change set via the GitHub git-data API, and there's no `sandbox-expired` status in the schema anymore. The `recordConfigAgentRunResult` here is now the "captured change missing" branch and returns `not-awaiting-review`. The described failure path is gone. Resolving.

File: apps/backend/src/lib/config/index.tsx [hunk 9]
```diff
@@ -475,6 +476,253 @@ export async function unlinkBranchConfigOverrideSource(options: {
...
+      return { nextRun: { status: "error", ...finished, error: options.outcome.error }, result: undefined };
+    }
+    if (options.outcome.status === "no-change") {
+      return { nextRun: { status: "no-change", ...finished }, result: undefined };
+    }
+    return {
+      nextRun: { status: "success", ...finished, commit_url: options.outcome.commitUrl },
+      // Advance the source's last-known commit when a commit landed and the branch
+      // is still GitHub-linked.
+      nextSource: source && options.outcome.newCommitHash ? { ...source, commit_hash: options.outcome.newCommitHash } : undefined,
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/backend/src/lib/config/index.tsx]: Confirmed — a real TOCTOU. On a successful commit the code advanced the source's `commit_hash` after re-reading the live branch source, checking only that its type was still `pushed-from-github`. If the branch was re-linked to a **different** repo mid-run, that new repo's source would inherit a commit SHA that only exists on the old repo. Fixed by threading the repo ref the commit was actually pushed against (`committedRef`) into `recordConfigAgentRunResult` and only advancing the hash when the live source's owner/repo/branch still matches it. Added a regression test for the re-link case. ✅ Fixed.

File: apps/dashboard/src/components/config-update/github-push-dialog.tsx [hunk 10]
```diff
@@ -0,0 +1,576 @@
...
+    if (configUpdate == null) {
+      onErrorChange("No configuration changes to push.");
+      return;
+    }
+    if (scopeCheck.status !== "ok") {
+      onErrorChange("Connect a GitHub account with the required scopes before pushing changes.");
+      return;
+    }
+    const adminInterface = getAdminInterface(adminApp);
+    if (adminInterface == null) {
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/dashboard/src/components/config-update/github-push-dialog.tsx:312]: Confirmed — and the root cause was a bit broader than the removed guard. The cancel handler put the dialog into a non-dismissible "cancelling" state and relied on the push poll loop to finish the transition, but that loop has already exited once the run reaches "awaiting_review" — so the dialog got stuck whether the cancel call threw **or** succeeded. Fixed by making the cancel handler settle the dialog itself (back to a dismissible state) after the cancel request completes, instead of depending on the poll loop. This covers all the stranding cases, including a missing method on an older build. ✅ Fixed.

File: apps/dashboard/src/components/config-update/github-push-dialog.tsx [hunk 11]
```diff
@@ -0,0 +1,576 @@
...
+    } catch (error) {
+      captureError("config-update-github-agent", {
+        projectId,
+        owner: source.owner,
+        repo: source.repo,
+        branch: source.branch,
+        configFilePath: source.configFilePath,
+        cause: error,
+      });
+      dialogContext?.setGithubRunActive(false);
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/dashboard/src/components/config-update/github-push-dialog.tsx]: Confirmed and fixed (same root cause as the cubic thread on this file). Clicking Discard during "awaiting_review" left the dialog permanently stuck in the non-dismissible "cancelling" state, because the only thing that settled it — the push poll loop — had already returned by that phase, so nothing observed the terminal status. The cancel handler now drives the terminal transition itself right after issuing the cancel, so the dialog always becomes dismissible again. ✅ Fixed.

File: packages/shared/src/schema-fields.ts [hunk 12]
```diff
@@ -947,3 +959,49 @@ export const branchConfigSourceSchema = yupUnion(
...
+
+/**
+ * State of a single dashboard→GitHub config agent run, so the dashboard can poll for
+ * progress and surface the resulting commit (or error). Each run is one row in the
+ * `ConfigAgentRun` table, addressed by `id`; runs are NOT serialized, so many can
+ * target the same branch at once and GitHub catches conflicts at push time.
+ */
+export const configAgentRunSchema = yupObject({
+  // The run's id (the `ConfigAgentRun` row id). The dashboard polls/cancels/commits this specific run by id.
+  id: yupString().defined(),
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply packages/shared/src/schema-fields.ts]: Agreed. The `ConfigAgentRun` id is always a DB-generated UUID (the column is `@db.Uuid`, and queries cast it to `::uuid`), and every other UUID row-id in this file already uses `.uuid()`. Tightened it to `yupString().uuid().defined()` to match. Backward-compatible with every real value. ✅ Fixed.

File: apps/dashboard/src/components/config-update/progress-content.tsx [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+ * monotonic delta and made the timer tick at ~2× speed.
+ */
+function useElapsedSeconds(startedAt: number): number {
+  const [monotonicElapsedMs, setMonotonicElapsedMs] = useState(0);
+  useEffect(() => {
+    const mountedAt = performance.now();
+    const t = setInterval(() => setMonotonicElapsedMs(performance.now() - mountedAt), 1000);
+    return () => clearInterval(t);
+  }, []);
+  const [initialOffsetMs] = useState(() => Math.max(0, currentEpochMsFromPerformance() - startedAt));
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/dashboard/src/components/config-update/progress-content.tsx]: Confirmed bug. The hook froze the run's start time at mount, so when a flow renders the progress box before the real start time is known (it starts at `0`, then is set a moment later), it subtracted `0` from the current epoch and showed a nonsensical ~57-year counter that never corrected. Rewrote `useElapsedSeconds` so its timer depends on `startedAt`: it recomputes the offset against a fresh monotonic anchor whenever `startedAt` changes, resetting to ~0 on that transition while still ticking accurately. ✅ Fixed.

**Comment by mantrakp04** [comment]
@greptile-ai review

File: packages/shared/src/config-eval.ts [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+  expect(() => evalConfigFileContent(`
+    import missingConfigPart from "./missing-config-part";
+    export const config = { auth: missingConfigPart };
+  `, "/tmp/hexclave-missing-import-config.ts")).toThrow();
+});
+
+import.meta.vitest?.test("evalConfigFileContent rejects syntactically invalid content", ({ expect }) => {
+  // jiti surfaces a ParseError (not a ConfigFileEvalError), so callers route this
+  // to "Failed to load config file" rather than "Invalid config".
+  expect(() => evalConfigFileContent("export const config = {", "stack.config.ts")).toThrow();
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply packages/shared/src/config-eval.ts]: Good point. The test documented a contract — a malformed file must surface as a *loader* error (a jiti ParseError), **not** a `ConfigFileEvalError`, so callers route it to "Failed to load config file" rather than "Invalid config" — but only asserted that it throws. Tightened it to also assert `not.toThrow(ConfigFileEvalError)`, so the error-class contract is actually protected if that routing ever regresses. ✅ Fixed.

File: apps/dashboard/src/components/config-update/progress-content.tsx [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+ * the box before its real start timestamp is known), so the counter resets
+ * instead of freezing a stale offset. Recomputing the offset every render —
+ * which is what the old code did — re-added the elapsed time on top of the
+ * monotonic delta and made the timer tick at ~2× speed.
+ */
+function useElapsedSeconds(startedAt: number): number {
+  const [elapsedMs, setElapsedMs] = useState(() => Math.max(0, currentEpochMsFromPerformance() - startedAt));
+  useEffect(() => {
+    const anchorPerfMs = performance.now();
+    const offsetMs = Math.max(0, currentEpochMsFromPerformance() - startedAt);
```
[the remark this answers is not in the document]
Comment (mantrakp04) [reply apps/dashboard/src/components/config-update/progress-content.tsx]: Confirmed — good catch on the first-paint flash. `startedAt=0` is the RDE dialog's "not started yet" sentinel; my earlier change made the counter react to `startedAt` changes but still computed `epoch − 0` (~56 years) on the very first paint, before the real timestamp lands. Added a `startedAt <= 0` guard so the counter reads 0 until a real start time arrives, then resets and ticks normally. ✅ Fixed.

**Comment by mantrakp04** [comment]
@greptile-ai review

**Comment by mantrakp04** [comment]
@greptile-ai review

File: apps/backend/src/lib/seed-dummy-data.ts [hunk 16]
```diff
@@ -2174,12 +2174,12 @@ export async function seedDummyProject(options: SeedDummyProjectOptions): Promis
...
-        owner: "stack-auth",
-        repo: "dummy-config-repo",
+        owner: "mantrakp04",
+        repo: "config-test",
         branch: "main",
         commit_hash: "abc123def456789",
-        config_file_path: "stack.config.json",
-        workflow_path: ".github/workflows/stack-auth-config-sync.yml",
+        config_file_path: "hexclave.config.ts",
+        workflow_path: ".github/workflows/hexclave-config-sync.yml",
```
Comment (N2D4) [inline apps/backend/src/lib/seed-dummy-data.ts]: we may want to change this to one that all of us have access to or so? `hexclave/dummy-config-repo` or so

File: apps/dashboard/src/components/config-update/rde-apply-dialog.tsx [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+import { currentEpochMsFromPerformance } from "./shared";
+
+type RdeApplyDialogProps = {
+  open: boolean,
+  adminApp: StackAdminApp<false> | null,
+  configUpdate: EnvironmentConfigOverrideOverride | null,
+  onSettle: (result: boolean) => void,
+};
+
+// CLI / remote-development-environment apply. Unlike the GitHub flow there's no
```
Comment (N2D4) [inline apps/dashboard/src/components/config-update/rde-apply-dialog.tsx:21]: PR looks mostly good but something about the onboarding is broken somehow, may have been there before but only visible now with this pr (would be nice if you could fix them though):  
  
\- when going through onboarding (you can test locally by changing hexclave.config.ts to `"show-onboarding"` and then going to http://127.0.0.1:9342/ ) at the end on the last step (2ith the Getting Started button) a "applying config changes" dialog pops up automatically without doing anything, i assume this is because it optimistically sets the config, but this feels weird, cuz i didn't do anything  
\- despite the first dialog, another dialog pops up when clicking Getting Started, and it seems this one never actually finishes. it seems it times out after a while. not sure why  
  
can you investigate?

**Review by N2D4 — APPROVED** [review]

File: apps/backend/src/lib/seed-dummy-data.ts (hunk 16 above)
Comment (mantrakp04) [reply apps/backend/src/lib/seed-dummy-data.ts]: we already do, its jus my github oauth app didnt have access to hexclave project for testing

File: apps/dashboard/src/components/config-update/rde-apply-dialog.tsx (hunk 17 above)
Comment (mantrakp04) [reply apps/dashboard/src/components/config-update/rde-apply-dialog.tsx:21]: Fixed it. The root cause was the welcome step pre-saving config in remote development environments: that opened the local “Apply configuration” dialog before the user clicked anything, then Get Started kicked off another save path. I changed project-onboarding-wizard.tsx so development environments only apply config from the final user action.

I also folded selected OAuth providers into the single development-environment config-file update and skipped the separate environment-only OAuth update there, which was the likely timeout/failure behind the second dialog. Added a regression in project-onboarding-wizard.test.tsx.
